Azure Pipelines is failing on Linux for both netstandard 1.4 and 2.0

I noticed the error in #3335 and also created the error on master by force building it. 

```
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018: The "GenerateDepsFile" task failed unexpectedly. [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018: System.MissingMethodException: Method not found: System.Collections.Generic.IList`1<NuGet.Packaging.Core.PackageDependency> NuGet.ProjectModel.LockFileTargetLibrary.get_Dependencies() [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at Microsoft.NET.Build.Tasks.DependencyContextBuilder+<>c__DisplayClass36_0.<GetLibraries>b__0 (NuGet.ProjectModel.LockFileTargetLibrary export) [0x00000] in <aafa1364e74c4a5ea60cdcaa95f71707>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Linq.Enumerable+SelectArrayIterator`2[TSource,TResult].MoveNext () [0x0003a] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Linq.Enumerable+<CastIterator>d__34`1[TResult].MoveNext () [0x0005f] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Collections.Generic.LargeArrayBuilder`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] items) [0x0003d] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Collections.Generic.SparseArrayBuilder`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] items) [0x00000] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Collections.Generic.SparseArrayBuilder`1[T].ReserveOrAdd (System.Collections.Generic.IEnumerable`1[T] items) [0x00017] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Linq.Enumerable+ConcatNIterator`1[TSource].LazyToArray () [0x00021] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Linq.Enumerable+ConcatNIterator`1[TSource].ToArray () [0x00008] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0001f] in <b5b8f552f9484fab8c19a20c404b4624>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at Microsoft.Extensions.DependencyModel.DependencyContext..ctor (Microsoft.Extensions.DependencyModel.TargetInfo target, Microsoft.Extensions.DependencyModel.CompilationOptions compilationOptions, System.Collections.Generic.IEnumerable`1[T] compileLibraries, System.Collections.Generic.IEnumerable`1[T] runtimeLibraries, System.Collections.Generic.IEnumerable`1[T] runtimeGraph) [0x00068] in <fa4af318e8374ad18eef8b73f115183f>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at Microsoft.NET.Build.Tasks.DependencyContextBuilder.Build () [0x001bd] in <aafa1364e74c4a5ea60cdcaa95f71707>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at Microsoft.NET.Build.Tasks.GenerateDepsFile.ExecuteCore () [0x00117] in <aafa1364e74c4a5ea60cdcaa95f71707>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at Microsoft.NET.Build.Tasks.TaskBase.Execute () [0x00000] in <aafa1364e74c4a5ea60cdcaa95f71707>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute () [0x00029] in <18acf4ba42414d46aa6ba7c7ca733b1f>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
/usr/share/dotnet/sdk/2.2.105/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:   at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask (Microsoft.Build.BackEnd.ITaskExecutionHost taskExecutionHost, Microsoft.Build.BackEnd.Logging.TaskLoggingContext taskLoggingContext, Microsoft.Build.BackEnd.TaskHost taskHost, Microsoft.Build.BackEnd.ItemBucket bucket, Microsoft.Build.BackEnd.TaskExecutionMode howToExecuteTask) [0x002a9] in <18acf4ba42414d46aa6ba7c7ca733b1f>:0  [/home/vsts/work/1/s/src/NUnitFramework/framework/nunit.framework.csproj]
```

[Full log](https://github.com/nunit/nunit/files/3464325/failingLinuxBuild.txt). A closely related (perhaps the same) issue has been reported in https://stackoverflow.com/questions/56613241/msbuild-generatedepsfile-fails

More related stuff:
* Incompatible versions of Mono MSBuild and .NET Core SDK (MSB4018) - https://rider-support.jetbrains.com/hc/en-us/articles/360004180039
* Building sdk projects with dotnet 2.2.x installed - `GeneratedDeps` failure - https://github.com/mono/mono/issues/13537
* Xamarin Build failed maybe nuget issue? - https://discuss.bitrise.io/t/xamarin-build-failed-maybe-nuget-issue/9535
* Problems with Rider 2019.1.2 - https://rider-support.jetbrains.com/hc/en-us/community/posts/360004228019-Problems-with-Rider-2019-1-2

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -27,6 +27,11 @@ jobs:
     vmImage: ubuntu-16.04
   steps:
   - bash: |
+     wget -q https://packages.microsoft.com/config/ubuntu/16.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+     sudo dpkg -i packages-microsoft-prod.deb
+     sudo apt-get install apt-transport-https
+     sudo apt-get update
+     sudo apt-get install dotnet-sdk-2.2
      dotnet tool install --global Cake.Tool
      export DOTNET_ROOT="$(dirname "$(readlink "$(command -v dotnet)")")"
      ~/.dotnet/tools/dotnet-cake --target=Test --test-run-name=Linux --configuration=Release

